app/src/main/java/ch/epfl/polychef/fragments/OnlineMiniaturesFragment.java

```diff
@@ -199,12 +195,30 @@ public boolean onClose() {
     private View.OnClickListener setFilter(int filter){
         return v -> {
             searchList.clear();
-            if(filter== FILTER_RECIPE){
+            if(filter == FILTER_RECIPE){
+                setAllBoolToFalse();
+                isFilterRecipe = true;
                 recipeStorage.getSearch().searchForRecipe(actualQuery, OnlineMiniaturesFragment.this);
-            }else if(filter== FILTER_USER){
+            }else if(filter == FILTER_USER){
+                setAllBoolToFalse();
+                isFilterUser = true;
                 userStorage.getSearch().searchForUser(actualQuery, OnlineMiniaturesFragment.this);
-            }else{
+            }else if (filter == FILTER_INGREDIENT){
+                setAllBoolToFalse();
+                isFilterIngredient = true;
                 recipeStorage.getSearch().searchRecipeByIngredient(actualQuery, OnlineMiniaturesFragment.this);
+            }else {
+                isFilterRate = true;
+                if(isFilterUser){
+                    userStorage.getSearch().searchForUser(actualQuery, OnlineMiniaturesFragment.this);
+                }else if(isFilterRecipe){
+                    recipeStorage.getSearch().searchForRecipe(actualQuery, OnlineMiniaturesFragment.this);
+                }else if(isFilterIngredient){
+                    recipeStorage.getSearch().searchRecipeByIngredient(actualQuery, OnlineMiniaturesFragment.this);
+                }else {
+                    userStorage.getSearch().searchForUser(actualQuery, OnlineMiniaturesFragment.this);
+                    recipeStorage.getSearch().searchForRecipe(actualQuery, OnlineMiniaturesFragment.this);
+                }
```

Comment:
If I understand correctly, `isFilterRate` is a toggle that we use at the same time as `isFilterRecipe` and `isFilterUser`?
Maybe the `onClick` should just be `isFilterRate = !isFilterRate` and then sort the existing result accordingly?
Perhaps we could make it a little bit more visual i.e. show which filter is currently on. But it should probably wait until next sprint